task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode3/ListNavFragment.kt

```diff
@@ -0,0 +1,79 @@
+package com.example.secondtasksolution.view.episode3
+
+import android.os.Build
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.View
+import androidx.activity.OnBackPressedCallback
+import androidx.fragment.app.setFragmentResultListener
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.secondtasksolution.R
+import com.example.secondtasksolution.adapter.CityAdapter
+import com.example.secondtasksolution.databinding.FragmentListNavBinding
+import com.example.secondtasksolution.model.City
+import com.example.secondtasksolution.util.CityDataSource
+import com.example.secondtasksolution.util.Constant.CITY_ID
+import com.example.secondtasksolution.util.Constant.REQUEST_KEY
+import com.example.secondtasksolution.util.Constant.UPDATED_CITY
+
+class ListNavFragment : Fragment(R.layout.fragment_list_nav) {
+
+    private var fragmentListNavBinding: FragmentListNavBinding? = null
+    private var listAdapter = CityAdapter(mutableListOf()) {}
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val binding = FragmentListNavBinding.bind(view)
+        fragmentListNavBinding = binding
+
+        val callback = object : OnBackPressedCallback(true) {
```

Comment:
Back aksiyonu için extension fonksiyonu yazabilir miyiz? `onBackPressed`, `handleBackPressed` olarak isimlendirilebilir. ActivityExt içerisine ekleyebilirsin. 

```
// Usage
onBackPressed {
  requireActivity().finish()
}
```